crates/router/src/services/authentication.rs

```diff
@@ -46,6 +46,14 @@ pub mod decision;
 pub struct AuthenticationData {
     pub merchant_account: domain::MerchantAccount,
     pub key_store: domain::MerchantKeyStore,
+    pub profile_id: Option<String>,
+}
+
+#[derive(Clone)]
+pub struct AuthenticationDataWithMultipleProfiles {
+    pub org_id: id_type::OrganizationId,
+    pub merchant_id: id_type::MerchantId,
```

Comment:
This should be merchant_account.
```suggestion
    pub merchant_account: domain::MerchantAccount,
```